tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -803,6 +804,22 @@ unique_ptr<DuckDBPyRelation> DuckDBPyConnection::ReadCSV(
 		bind_parameters["delim"] = Value(py::str(delimiter));
 	}
 
+	if (!py::none().is(auto_type_candidates)) {
+		if (!py::isinstance<py::list>(auto_type_candidates)) {
+			throw InvalidInputException("read_csv only accepts 'auto_type_candidates' as a list of types");
+		}
+		vector<Value> candidates;
+		py::list candidates_list = auto_type_candidates;
+		for (auto &elem : candidates_list) {
+			shared_ptr<DuckDBPyType> sql_type;
+			if (!py::try_cast(elem, sql_type)) {
+				throw py::value_error("The types provided to 'auto_type_candidates' have to be DuckDBPyType");
+			}
+			candidates.push_back(sql_type->ToString());
+		}
```

Comment:
This looks rather similar to the code that checks the `dtype` option, maybe it could be turned into a method?